lib/dev/ui/spinner.rb

```diff
@@ -62,8 +63,16 @@ def check
           end
 
           def render(index, force = true)
-            return full_render(index) if force
+            return full_render(index) if force || @force_full_render
             partial_render(index)
+          ensure
+            @force_full_render = false
+          end
+
+          def update_title(new_title, erase_rest_of_line: true)
+            @title = new_title
+            @title += "\e[K" if erase_rest_of_line
```

Comment:
IMO this doesn't need to be optional, but I think this is the wrong place to do it. It's not really part of the title. I think we could just always print "\e[K" at the end of titles, and then forcing a full render is sufficient.